controller/cache/cache.go

```diff
@@ -824,6 +829,79 @@ func (c *liveStateCache) handleAddEvent(cluster *appv1.Cluster) {
 	}
 }
 
+// namespaceDiff calculates the added and removed namespaces between old and new lists.
+func namespaceDiff(oldNamespaces, newNamespaces []string) (added, removed []string) {
+	oldNsSet := make(map[string]bool)
+	for _, ns := range oldNamespaces {
+		oldNsSet[ns] = true
+	}
+
+	newNsSet := make(map[string]bool)
+	for _, ns := range newNamespaces {
+		newNsSet[ns] = true
+	}
+
+	// Find added namespaces
+	for _, ns := range newNamespaces {
+		if !oldNsSet[ns] {
+			added = append(added, ns)
+		}
+	}
+
+	// Find removed namespaces
+	for _, ns := range oldNamespaces {
+		if !newNsSet[ns] {
+			removed = append(removed, ns)
+		}
+	}
+
+	return added, removed
+}
+
+// handleNamespaceChanges processes namespace changes and returns update settings if needed.
+// Returns nil if no updates are required (incremental sync succeeded or no changes).
+// Returns update settings for full invalidation if incremental sync is disabled or failed.
+func (c *liveStateCache) handleNamespaceChanges(
+	cluster clustercache.ClusterCache,
+	oldNamespaces, newNamespaces []string,
+) []clustercache.UpdateSettingsFunc {
+	if reflect.DeepEqual(oldNamespaces, newNamespaces) {
```

Comment:
`reflect.DeepEqual` will return false if the same namespaces are providedd, but in different order. It is better to use semantic equality check, Consider using `k8s.io/apimachinery/pkg/api/equality.Semantic.DeepEqual` https://pkg.go.dev/k8s.io/apimachinery/pkg/api/equality